carbon-design-system/carbon-addons-iot-react
# fix(table): column resize failing for initially hidden column with un…

Closes #1419 

**Summary**
Table resizing doesn't work when the user toggles the visibility of an initially hidden column that did not have a width property in the columns prop.

**Change List (commits, features, bugs, etc)**
- The fix was to create a new width (based on a column average) in the case a toggled column that should be shown did not have an existing width property.
- Refactored the `calculateWidthOnShow` and its sub functions for improved readability
- added a unit test to prevents future regression 

**Acceptance Test (how to verify the PR)**

1. go to https://deploy-preview-1420--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table--stateful-example-with-expansion-maxpages-and-column-resize
2. Click on "Column selection" button
3. Click on "Secret information field" to toggle the visibility
4. Verify that the "Secret information" column becomes visible with a width that is close to the average column widths
5. Verify that the column resize is still working properly
